CyclopsNuclearReactor/CyNukReactorSMLHelper.cs

```diff
@@ -0,0 +1,211 @@
+
+            return prefab;
+        }
+
+        /// <summary>
+        /// Applies the shader to the materials of the reactor
+        /// </summary>
+        /// <param name="prefab"></param>
+        private void ApplyShaders(GameObject prefab)
+        {
+            #region SystemLights_BaseColor
+            MaterialHelpers.ApplyEmissionShader("SystemLights_BaseColor", "SystemLights_OnMode_Emissive", prefab, _assetBundle, new Color(0.08235294f, 1f, 1f));
+            MaterialHelpers.ApplyNormalShader("SystemLights_BaseColor", "SystemLights_Norm", prefab, _assetBundle);
+            MaterialHelpers.ApplyAlphaShader("SystemLights_BaseColor", prefab);
+            #endregion
+
+            #region FCS_SUBMods_GlobalDecals
+            MaterialHelpers.ApplyAlphaShader("FCS_SUBMods_GlobalDecals", prefab);
+            #endregion
+
+            #region NuclearPowerStorage
+            MaterialHelpers.ApplyEmissionShader("NuclearPowerStorage", "NuclearPowerStorage_Emissive", prefab, _assetBundle, new Color(0.08235294f, 0.7686275f, 0.227451f));
+            MaterialHelpers.ApplyNormalShader("NuclearPowerStorage", "NuclearPowerStorage_Norm", prefab, _assetBundle);
+            #endregion
+
+            #region Glass
+            // MaterialHelpers.ApplyGlassShader("glass", prefab, 0.1f); //Temperately disabled until solution is found
+            #endregion
+        }
+
+
+        /// <summary>
+        /// Finds the prefab in the asset bundle
+        /// </summary>
+        /// <returns></returns>
+        public bool GetPrefabs()
+        {
+            // == Get the prefab == //
+
+            AssetBundle assetBundle = AssetHelper.Asset("CyclopsNuclearReactor", "CyNukReactorbundle");
+
+            //If the result is null return false.
+            if (assetBundle == null)
+                return false;
+
+            _assetBundle = assetBundle;
+
+            //We have found the asset bundle and now we are going to continue by looking for the model.
+            GameObject cyNukReactorPrefab = assetBundle.LoadAsset<GameObject>("CyNukReactor");
+
+            //If the prefab isn't null lets add the shader to the materials
+            if (cyNukReactorPrefab != null)
+            {
+                _cyNukReactorPrefab = cyNukReactorPrefab;
+
+                //Lets apply the material shader
+                ApplyShaders(_cyNukReactorPrefab);
+            }
+            else
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        protected override TechData GetBlueprintRecipe()
+        {
+            return new TechData
+            {
+                Ingredients =
+                {
+                    new Ingredient(TechType.PlasteelIngot, 1),
+                    new Ingredient(TechType.AdvancedWiringKit, 1),
+                    new Ingredient(TechType.Lead, 2),
```

Comment:
Any chance we can make this recipe more interesting?  
A nuclear reactor in a base also requires the cost of the Multi-Purpose Room it goes inside.  
We might want to make this just a _little_ more expensive.

Reply:
Instead of over-complicating the recipe for the reactor itself, we will instead introduce a pair of tiered upgrade modules to enhance the reactor by expanding its capacity.  
Due to how reactor rods are processed in parallel by the reactor, adding more rods with naturally increase the charging rate of the reactor.

The upgrade modules will be modeled after the [BioReactorBooster](https://github.com/PrimeSonic/PrimeSonicSubnauticaMods/blob/master/MoreCyclopsUpgrades/CyclopsUpgrades/BioBoosterUpgradeHandler.cs) as their function is very similar.

Reply:
Feature implemented. Needs to be tested and have the recipes looked over.